tests/unit/test_tiler.py

```diff
@@ -778,6 +779,73 @@ def it_knows_whether_coordinates_are_within_extraction_mask(
         assert type(coords_within_extraction_mask) == bool
         assert coords_within_extraction_mask == expected_result
 
+    @pytest.mark.parametrize(
+        "outline, n_tiles, error_msg",
+        (
+            (
+                ["yellow", "yellow"],
+                3,
+                "There should be as many outlines as there are tiles!",
+            ),
+            (
+                ["yellow", "yellow"],
+                1,
+                "There should be as many outlines as there are tiles!",
+            ),
+            (
+                0.5,
+                2,
+                "The parameter ``outline`` should be of type: "
+                "str, Iterable[str], or Iterable[List[int]]",
+            ),
+        ),
+    )
+    def it_throws_error_with_invalid_tile_outline(
+        self, request, tmpdir, outline, n_tiles, error_msg
+    ):
+        slide, _ = base_test_slide(tmpdir, PILIMG.RGBA_COLOR_500X500_155_249_240)
+        coords = CP(0, 10, 0, 10)
+        tile = Tile(PILIMG.RGBA_COLOR_500X500_155_249_240, coords)
+        mock_tiles = [(tile, coords)] * n_tiles
+        _tiles_generator = method_mock(request, GridTiler, "_tiles_generator")
+        _tiles_generator.return_value = mock_tiles
+        grid_tiler = GridTiler((10, 10))
+
+        with pytest.raises(ValueError) as err:
+            grid_tiler.locate_tiles(slide=slide, outline=outline)
+
+        assert str(err.value) == error_msg
+
+    @pytest.mark.parametrize(
+        "pass_tiles, outline, expected_topleft",
+        (
+            (False, "red", (255, 255, 155, 155)),
+            (True, ["red", "red"], (255, 255, 155, 155)),
+            (True, [(255, 0, 0), (255, 0, 0)], (255, 255, 155, 155)),
+        ),
+    )
+    def it_can_locate_tiles(
+        self, request, tmpdir, pass_tiles, outline, expected_topleft
+    ):
+        slide, _ = base_test_slide(tmpdir, PILIMG.RGBA_COLOR_500X500_155_249_240)
+        coords = CP(0, 10, 0, 10)
+        tile = Tile(PILIMG.RGBA_COLOR_500X500_155_249_240, coords)
+        mock_tiles = [(tile, coords), (tile, coords)]
+        method_mock(request, GridTiler, "_tiles_generator", return_value=mock_tiles)
+        method_mock(
+            request,
+            GridTiler,
+            "_grid_coordinates_generator",
+            return_value=[CP(0, 10, 0, 10), CP(0, 10, 0, 10)],
+        )
```

Comment:
why do we need this? this method is not called in the `locate_tiles` method, and we mock `_tiles_generator` that would call it

moreover if we mock something it's also better to assert that the mock has been called

Reply:
> why do we need this? this method is not called in the `locate_tiles` method, and we mock `_tiles_generator` that would call it
yes u right this is not needed
> 
> moreover if we mock something it's also better to assert that the mock has been called
that's not true Alessia, if we mock someting for the sake of isolation it's not mandatory to assert that something has been called ... we mock a lot of property for example and we do not assert anything 🤷🏽‍♂️